dmutils/forms/fields.py

```diff
@@ -0,0 +1,35 @@
+from itertools import chain
+
+from wtforms import StringField
+from wtforms.validators import Length
+
+from .validators import EmailValidator
+
+
+class StripWhitespaceStringField(StringField):
+    def __init__(self, label=None, **kwargs):
+        kwargs['filters'] = tuple(chain(
+            kwargs.get('filters', ()),
+            (
+                strip_whitespace,
+            ),
+        ))
+        super(StringField, self).__init__(label, **kwargs)
+
+
+class EmailField(StripWhitespaceStringField):
+    def __init__(self, label=None, **kwargs):
+        kwargs["validators"] = tuple(chain(
+            kwargs.pop("validators", ()),
+            (
+                EmailValidator(),
+                Length(max=511, message="Please enter an email address under 512 characters."),
+            ),
+        ))
+        super(EmailField, self).__init__(label, **kwargs)
+
+
+def strip_whitespace(value):
```

Comment:
Running `ag` on my checkouts shows no uses outside of dmutils, but there might be some repos that I am missing.

Could be ultra-cautious and add it just to avoid breaking changes, thoughts?

Reply:
Do you want to rename it here with an `_` in that case? Don't mind either way.

Reply:
After thinking about it, I decided that actually it is a useful filter that *should* be usable outside of the package, so I put it into it's own module.